### PR TITLE
Refactor child compiler to remove deprecation warnings

### PR DESCRIPTION
**Pull request by mattcompiles**

### CODE REVIEW

File: packages/treat/src/webpack-plugin/treatCompiler.js [hunk 1]
```diff
@@ -106,48 +99,86 @@ function getRootCompilation(loader) {
...
-        new NodeTemplatePlugin(outputOptions),
-        new LibraryTemplatePlugin(null, 'commonjs2'),
-        new NodeTargetPlugin(),
-        new SingleEntryPlugin(loader.context, loader.resourcePath),
-        new LimitChunkCountPlugin({ maxChunks: 1 }),
-        new ExternalsPlugin('commonjs', 'treat'),
-      ],
     );
 
-    const subCache = 'subcache ' + __dirname + ' ' + loader.resourcePath;
```
Comment (markdalgleish) [inline packages/treat/src/webpack-plugin/treatCompiler.js:127]: This cache doesn't seem to have been ported across?

File: packages/treat/src/webpack-plugin/treatCompiler.js [hunk 2]
```diff
@@ -106,48 +99,86 @@ function getRootCompilation(loader) {
...
+              compilation.deleteAsset(file);
+            });
+          });
         });
       });
-
-      callback();
-    });
+    } else {
+      childCompiler.hooks.afterCompile.tap(TWL, (compilation) => {
```
Comment (markdalgleish) [inline packages/treat/src/webpack-plugin/treatCompiler.js:169]: This is no longer async?

**Comment by loliver** [comment]
Any chance this can be looked at? Running this branch via `npm link` fixes our Playroom build on our Treat tech stack. Previously was getting a `TypeError: The 'compilation' argument must be an instance of Compilation` error within each component's `treat.ts` file.

We also have an instance of Playroom running fine in the same codebase with our old `styled-components` based stack from the same `node_modules` folder without issue.

File: packages/treat/src/webpack-plugin/treatCompiler.js (hunk 1 above)
Comment (mattcompiles) [reply packages/treat/src/webpack-plugin/treatCompiler.js:127]: This was intentional. Not needed anymore.

File: packages/treat/src/webpack-plugin/treatCompiler.js (hunk 2 above)
Comment (mattcompiles) [reply packages/treat/src/webpack-plugin/treatCompiler.js:169]: Doesn't need to be. Probably a mistake in the first place.

**Review by markdalgleish — APPROVED** [review]

**Comment by mattcompiles** [comment]
@loliver Hey Bud. Sorry this one slipped through the cracks while we were working on vanilla-extract. Released in `v2.0.3`.

**Comment by loliver** [comment]
> @loliver Hey Bud. Sorry this one slipped through the cracks while we were working on vanilla-extract. Released in `v2.0.3`.

No problems at all! Thanks for getting the quick release done this morning.

**Pull request merged**